src/toolkit/res_map.h

```diff
@@ -233,7 +233,7 @@ class ResMap {
   }
 
   /// Whether quantity_ should be recomputed or not.
-  bool dirty_quantity_;
+  mutable bool dirty_quantity_;
```

Comment:
We may have discussed this before, but why do we suddenly need this? And only here?

Reply:
At line 100 we declare a const function which guarantees members of the class will not be modified.  In that function `dirty_quantity_` is modified, and clang throws a fatal error:
 ```clang
error: cannot assign to non-static data member within const member function 'operator[]'
  101 |     dirty_quantity_ = true;

 ```
 
 Marking `dirty_quantity_` as `mutable` allows us to modify it as we please.

Reply:
I guess it's not really a `const` function then... but I see why it needs to exist, so maybe that's why we have `mutable` to allow us to lie about the `const`-ness of a function 🤣